### PR TITLE
Cmis

### PR DESCRIPTION
**Pull request by bwrsandman**
Port of https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis/+merge/212258

This module allows to configure a CMIS backend in OpenERP.
## LP reviews:
- @hbrunn: Approve on 2014-04-07 
- @max3903: Needs Fixing on 2014-04-24
  
  > @ehdem
  > l205: Can you put CMIS versions (1.0 or 1.1, not sure) ? 1.7 is a Magento version.
  > Thanks
- @bwrsandman: Needs Information on 2014-05-14 
  
  > Looking a lot better, still issues about the comments preceding functions which would be more useful as docstrings.

### CODE REVIEW

File: cmis/cmis_model.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                      "Please remove it and try again."))
+            except cmislib.exceptions.RuntimeException:
+                raise orm.except_orm(
+                    _('Cmis access right Error!'),
+                    ("Please check your access right."))
+        self.get_error_for_path(bool_path_write, folder_path_write)
+
+    # Function to check if we have access right to read from the path
+    def check_directory_of_read(self, cr, uid, ids, context=None):
+        ir_attach_obj = self.pool.get('ir.attachment')
```
Comment (bwrsandman) [inline cmis/cmis_model.py:104]: Flake8: `F841 local variable 'ir_attach_obj' is assigned to but never used`

File: cmis/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Contributors
+------------
+* El Hadji Dem (elhadji.dem@savoirfairelinux.com)
+""",
+    'author': 'Savoir-faire Linux',
+    'website': 'www.savoirfairelinux.com',
+    'license': 'AGPL-3',
+    'depends': [
+        'connector',
+    ],
```
Comment (bwrsandman) [inline cmis/__openerp__.py:49]: There should be an `external_dependencies` for `cmislib`

File: cmis/connector.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+def get_environment(session, model_name, backend_id):
+    """ Create an environment to work with.  """
+    backend_record = session.browse('cmis.backend', backend_id)
+    env = Environment(backend_record, session, model_name)
+    lang = backend_record.default_lang_id
+    lang_code = lang.code if lang else 'en_US'
+    env.set_lang(code=lang_code)
+    return env
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (bwrsandman) [inline cmis/connector.py:38]: Please remove vim lines.

File: cmis/backend.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import openerp.addons.connector.backend as backend
+
+
+cmis = backend.Backend('cmis')
+""" Generic CMIS Backend """
+
+cmis1000 = backend.Backend(parent=cmis, version='1.0')
+""" CMIS Backend for version 1.0 """
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (bwrsandman) [inline cmis/backend.py:32]: Please remove vim lines.

File: cmis/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    'js': [],
+    'qweb': [],
+    'test': [],
+    'demo': [],
+    'installable': True,
+    'auto_install': False,
+}
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (bwrsandman) [inline cmis/__openerp__.py:62]: Please remove vim lines.

File: cmis/__init__.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from . import cmis_model
+from . import backend
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (bwrsandman) [inline cmis/__init__.py:26]: Please remove vim lines.

**Comment by bwrsandman** [comment]
This MP should also add `- sudo pip install cmislib` to the `.travis.yml` `install` section

**Comment by max3903** [comment]
This MP should be in the OCA/connector-cmis, not OCA/knowledge

**Pull request closed without merging**